packages/web3torrent/src/clients/web3t-channel-client.ts

```diff
@@ -0,0 +1,193 @@
+import {ChannelResult, Message, ChannelClient} from '@statechannels/channel-client';
+import {bigNumberify} from 'ethers/utils';
+
+interface ChannelState {
+  channelId: string;
+  turnNum: string;
+  challengeExpirationTime;
+  aUserId: string;
+  bUserId: string;
+  aAddress: string;
+  bAddress: string;
+  aOutcomeAddress: string;
+  bOutcomeAddress: string;
+  aBal: string;
+  bBal: string;
+}
+
+// This class wraps the channel client converting the request/response formats to those used in the app
+
+export class Web3TorrentChannelClient {
+  constructor(private readonly channelClient: ChannelClient) {}
+
+  mySigningAddress?: string;
+  myEthereumSelectedAddress?: string; // this state can be inspected to infer whether we need to get the user to "Connect With MetaMask" or not.
+
+  async enable() {
+    /* empty */
+  }
+
+  async createChannel(
+    aAddress: string,
+    bAddress: string,
+    aBal: string,
+    bBal: string,
+    aOutcomeAddress: string = aAddress,
+    bOutcomeAddress: string = bAddress
+  ): Promise<ChannelState> {
+    const participants = formatParticipants(aAddress, bAddress, aOutcomeAddress, bOutcomeAddress);
+    const allocations = formatAllocations(aOutcomeAddress, bOutcomeAddress, aBal, bBal);
+    const appDefinition = '0x0'; // TODO SingleAssetPayments address
```

Comment:
Maybe stick with `AddressZero` and reference the fact that this is the null app in a comment in the code

Reply:
I think we have some freedom in whether we use the null app or the single assets payments app, but the choice will have implications on how the wallet should behave. I think this placeholder is probably ok until we decide which we want to do.